api/sip/client.py

```diff
@@ -295,7 +295,14 @@ def connect(self):
         """Create a socket connection to a SIP server."""
         with self.socket_lock:
             sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-            sock.connect((self.target_server, self.target_port))
+            try:
+                sock.connect((self.target_server, self.target_port))
+            except TypeError:
+                raise IOError(
+                    "Could not connect to %s:%s" % (
+                        self.target_server, self.target_port
+                    )
+                )
```

Comment:
This was necessary for the test to work without a lot of extra scaffolding, but it also handles the case where you've created a SIP2 server and given it absolutely no configuration. Any IOError is turned into a RemoteIntegrationException in SIP2AuthenticationProvider.__init__, but TypeError is such a generic error that I don't want to catch all of them.